test/widget/tool/hand_tool_test.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:paintroid/service/device_service.dart';
+import 'package:paintroid/tool/tool.dart';
+import 'package:paintroid/ui/pocket_paint.dart';
+
+import '../utils/utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  late Widget sut;
+  double epsilon = 0.1;
+
+  setUp(() {
+    sut = ProviderScope(
+      overrides: [
+        IDeviceService.sizeProvider
+            .overrideWith((ref) => Future.value(const Size(600, 600)))
+      ],
+      child: const MaterialApp(
+        home: PocketPaint(),
+      ),
+    );
+  });
+
+  testWidgets('Pan bottom left', (WidgetTester tester) async {
+    await tester.pumpWidget(sut);
+    await tester.pumpAndSettle();
+
+    final bottomNavBarInteractions = BottomNavBarInteractions(tester);
+    await bottomNavBarInteractions.selectTool(ToolData.HAND);
+
+    final finder = find.byType(InteractiveViewer);
+
+    expect(finder, findsOneWidget);
+
+    InteractiveViewer interactiveViewer = tester.widget(finder);
+
+    TransformationController controller =
+        interactiveViewer.transformationController!;
+
+    expect(controller, isNotNull);
+
+    final initialMatrix = controller.value;
+
+    await tester.drag(finder, const Offset(-50, 50));
+    await tester.pumpAndSettle();
+
+    double expectedX = initialMatrix.getTranslation().x - 50;
+    double expectedY = initialMatrix.getTranslation().y + 50;
+
+    expect(controller.value.getTranslation().x, closeTo(expectedX, epsilon));
+    expect(controller.value.getTranslation().y, closeTo(expectedY, epsilon));
```

Comment:
The code is always the same except for the x and y translation, we should write a function for that, instead of having duplicate code

Reply:
Nice catch 👍🏻  wrote a function for that.